src/payment/models.py

```diff
@@ -0,0 +1,55 @@
+from django.db import models
+
+from nurse.models import User
+
+
+class Subscription(models.Model):
+    user = models.OneToOneField(
+        User, on_delete=models.CASCADE, related_name="subscription"
+    )
+    stripe_subscription_id = models.CharField(max_length=255, unique=True)
+    status = models.CharField(max_length=50, null=True)
+    payment_status = models.CharField(max_length=50, null=True)
+    plan = models.BooleanField(default=False)
+    product_name = models.CharField(max_length=255, null=True, blank=True)
+    current_period_start = models.DateTimeField(null=True, blank=True)
+    current_period_end = models.DateTimeField(null=True, blank=True)
+    cancel_at_period_end = models.BooleanField(default=False)
+    trial_end = models.DateTimeField(null=True, blank=True)
+    total_price = models.DecimalField(
+        max_digits=10, decimal_places=2, null=True, blank=True
+    )
+    hosted_invoice_url = models.URLField(max_length=500, null=True, blank=True)
+    invoice_pdf = models.URLField(max_length=500, null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f"Subscription of {self.user.username}"
+
+
+class CustomerDetail(models.Model):
+    user = models.OneToOneField(
+        User, on_delete=models.CASCADE, related_name="customer_details"
+    )
+    stripe_customer_id = models.CharField(max_length=255, unique=True)
+    city = models.CharField(max_length=100, null=True, blank=True)
+    country = models.CharField(
+        max_length=2, null=True, blank=True
+    )  # Code ISO countries
+    address = models.CharField(max_length=255, null=True, blank=True)
+    postal_code = models.CharField(max_length=20, null=True, blank=True)
+    email = models.EmailField(null=True, blank=True)
```

Comment:
Yes, you're right that some fields might seem redundant. However, the documentation above explains why we keep the email in the customer table. It's because we need to maintain a direct link to the customer's contact information for billing and other purposes, separate from the user/profile data, which is used for authentication and user-specific settings.